app/controllers/api/v1/user_organizations_controller.rb

```diff
@@ -0,0 +1,51 @@
+#:reek:InstanceVariableAssumption
+#:reek:TooManyInstanceVariables
+class Api::V1::UserOrganizationsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_current_user_organization, only: %i[index]
+  before_action :set_user_organization, only: %i[destroy]
+  before_action :set_organization, only: %i[destroy]
+  before_action :set_user, only: %i[destroy]
+
+  def destroy
+    if current_user.organization_member?(@organization.id) && current_user.id == @organization.owner_id
```

Comment:
can we make logic here simpler by moving  to service?
I just want to see here destroy or not without such list of conditions